### ISSUE TITLE
Crashing on launch since v2

### ISSUE THREAD

**Issue by 1xPdd** [iSoron/uhabits#1399]
## Pre-submission checklist
- [x] I am submitting a bug report, not a feature request.
- [x] I have searched for similar issues, but did not find any matches.

## Description
Since v2.0, Loop Habits crashes whenever I start the app. Yet for habits with notifications I am able to mark those as complete and, if I select the notification, I'm taken to the detail view for that specific habit (sorry, I don't know how to describe it. It has various graphs of habit's history). Tracking these continue to work and the data appears to be saved and updated from day to day.

## Steps to reproduce
1. Launch the app using default Android launcher

## System information
 - Phone: Samsung Galaxy S5
 - Phone Operating System: LineageOS 14 (Android 7.1.2)
 - App version: v2.0.3 from F-Droid

## Logs
```
[06-03 07:46:52.944 8038:8038 I/DatabaseUtils]
Writing: /storage/emulated/0/Android/data/org.isoron.uhabits/files/Backups/Loop Habits Backup 2022-06-03 074652.db

[06-03 07:46:52.947 8038:8038 W/System.err]
java.lang.RuntimeException: Unable to resume activity {org.isoron.uhabits/org.isoron.uhabits.MainActivity}: java.io.FileNotFoundException: /data/user/0/org.isoron.uhabits/files/../databases/uhabits.db (Not a directory)

[06-03 07:46:52.948 8038:8038 W/System.err]
Caused by: java.io.FileNotFoundException: /data/user/0/org.isoron.uhabits/files/../databases/uhabits.db (Not a directory)

[06-03 07:46:52.949 8038:8038 W/System.err]
	at org.isoron.uhabits.utils.FileUtilsKt.copyTo(FileUtils.kt:30)

[06-03 07:46:52.949 8038:8038 W/System.err]
	at org.isoron.uhabits.utils.DatabaseUtils.saveDatabaseCopy(DatabaseUtils.kt:67)

[06-03 07:46:52.949 8038:8038 W/System.err]
	at org.isoron.uhabits.database.AutoBackup.run(AutoBackup.kt:43)

[06-03 07:46:52.949 8038:8038 W/System.err]
	at org.isoron.uhabits.database.AutoBackup.run$default(AutoBackup.kt:33)

[06-03 07:46:52.949 8038:8038 W/System.err]
	at org.isoron.uhabits.activities.habits.list.ListHabitsActivity.onResume(ListHabitsActivity.kt:91)

[06-03 07:46:53.084 8038:8038 E/AndroidRuntime]
FATAL EXCEPTION: main
Process: org.isoron.uhabits, PID: 8038
java.lang.RuntimeException: Unable to resume activity {org.isoron.uhabits/org.isoron.uhabits.MainActivity}: java.io.FileNotFoundException: /data/user/0/org.isoron.uhabits/files/../databases/uhabits.db (Not a directory)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3454)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3494)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2757)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6186)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
Caused by: java.io.FileNotFoundException: /data/user/0/org.isoron.uhabits/files/../databases/uhabits.db (Not a directory)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at org.isoron.uhabits.utils.FileUtilsKt.copyTo(FileUtils.kt:30)
	at org.isoron.uhabits.utils.DatabaseUtils.saveDatabaseCopy(DatabaseUtils.kt:67)
	at org.isoron.uhabits.database.AutoBackup.run(AutoBackup.kt:43)
	at org.isoron.uhabits.database.AutoBackup.run$default(AutoBackup.kt:33)
	at org.isoron.uhabits.activities.habits.list.ListHabitsActivity.onResume(ListHabitsActivity.kt:91)
	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1270)
	at android.app.Activity.performResume(Activity.java:6788)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3431)
	... 10 more

[06-03 07:46:53.088 6066:22301 W/ActivityManager]
  Force finishing activity org.isoron.uhabits/.MainActivity

[06-03 07:46:53.595 6066:6092 W/ActivityManager]
Activity pause timeout for ActivityRecord{9fd98b7 u0 org.isoron.uhabits/.MainActivity t58099 f}

[06-03 07:46:55.254 6066:22301 W/ActivityManager]
  Force finishing activity org.isoron.uhabits/.MainActivity

[06-03 07:46:55.255 6066:22301 I/ActivityManager]
Killing 8038:org.isoron.uhabits/u0a84 (adj 900): crash

[06-03 07:46:55.274 6066:6093 W/ViewRootImpl[uhabits]]
Dropping event due to root view being removed: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=912.0, y[0]=-205.0, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=1082112196, downTime=1082112185, deviceId=16, source=0x5002 }
```

Data appears to be saved here: `/storage/emulated/0/Android/data/org.isoron.uhabits/files/` which shows Logs, CSV, and Backups directories, each filled with various files for the last ~5 years.

**Comment by hiqua** [comment]
Are you able to test that without any data? I.e. what would happen if the app just got freshly installed?

**Comment by 1xPdd** [comment]
I've been having a bit of trouble export all of up my habit data, since I can't open the program. Similarly, `adb backup` produces a 47 byte file, so I was a bit hesitant to wipe things. I found a program called oandbackup, made a backup and cleared data & cache for Habits. I opened it, went through the introductory tour, and was able to use it without problem. Strangely, I stopped it and restored the backup, then opened Habits again: it worked fine and, as far as I can tell, I have all my old data. I'm not sure what changed in the backup/restore, but I guess this is fixed. Thanks.
